counterpartylib/lib/messages/issuance.py

```diff
@@ -124,6 +137,36 @@ def validate (db, source, destination, asset, quantity, divisible, callable_, ca
         if destination:
             problems.append('cannot transfer a non‐existent asset')
 
+    # validate parent ownership for subasset
```

Comment:
this would also catch the (infinitesimally small) chance that `util.generate_random_asset()` in `compose` generates a duplicate asset ID already used by the same source address for another asset